### ISSUE TITLE
GraphQL response is `str` not `dict`

### ISSUE THREAD

**Issue by DillonB07** [DillonB07/GitStats#3]
The response returned by the `requests.post()` is of type `str` not `dict` meaning it can't be iterated through.

**Comment by DillonB07** [comment]
@JBYT27 did you encounter this issue in your project? If you did, what was the solution for it? I had a look through your code but couldn't see anything to fix it.

**Comment by JBYT27** [comment]
Maybe try removing `request.json()` on the following line:
https://github.com/DillonB07/GitStats/blob/0b825f89f28598a462361a95100cedc8f0cf2e5a/src/pygitstats/graphql/utils.py#L17
and replacing it with `request.json`.

**Comment by DillonB07** [comment]
No, that gives an error. `json()` is a function in the `request` class.

**Comment by JBYT27** [comment]
Oh, I can't check right now, but is the request graphql function the same as on my module?

**Comment by DillonB07** [comment]
Pretty much. I modifed it slightly but the actual request handling is the same:
```py
def get_query(headers, query):
    request = requests.post(URL, json={'query': query}, headers=headers)
    if request.status_code == 200:
        return json.dumps(request.json(), indent=4, sort_keys=True)
    else:
        raise QueryFailError("Query failed to run by returning code of {}. {}".format(
            request.status_code, query))
```

**Comment by JBYT27** [comment]
Ah, I think I know what's happening. So the query below should be a `dict`, not a string. Then it should return the correct value as a `dict`, and return custom queries as well.
```py
query = '''
query {
  user(login: "''' + user + '''") { 
    url
    login
    name
    twitterUsername
    websiteUrl
    email
    company
    location
    status {
      message
      indicatesLimitedAvailability
    }
    avatarUrl
    bio
    isDeveloperProgramMember
    isCampusExpert
    starredRepositories {
      totalCount
    }
    followers {
      totalCount
    }
    following {
      totalCount
    }
    issues {
      totalCount
    }
    pullRequests {
      totalCount
    }
    itemShowcase {
      hasPinnedItems
      items(first: 10) {
        nodes {
          ... on Repository {
            name
          }
        }
      }
    }
    organizations(first: 10) {
      totalCount
      nodes {
        name
      }
    }
    repositories {
      totalCount
    }
  }
}
        '''
        response = get_query(self.headers, query)
        return response
```

**Comment by DillonB07** [comment]
Thanks, but that also doesn't work as it's invalid syntax for a dictionary. I've found a module called `gql` which should hopefully work.

**Comment by JBYT27** [comment]
Oh, okay! Just compared my module with yours, but maybe you forgot to actually name your query? Like here:
https://github.com/JBYT27/Git-API/blob/6bcbb6bb69b73f1f6979ed79568c3dad547de1ef/src/git_api/main.py#L84
and then here:
https://github.com/DillonB07/GitStats/blob/f91db82feef6fdf585422c923602fc43e8af6cef/src/pygitstats/graphql/gitstats.py#L19

**Comment by DillonB07** [comment]
That doesn't make a difference. I checked it with the [GraphiQL explorer](https://docs.github.com/en/graphql/overview/explorer) and the script

**Comment by DillonB07** [comment]
The [`graphql-python/gql`](https://github.com/graphql-python/gql) seems to work fine.

There was probably a much easier way to do it, but I've not a clue why the response was returning as a string.

**Comment by JBYT27** [comment]
Oh, okay. Sorry that I couldn't help!

**Comment by DillonB07** [comment]
No problem!

**Comment by DillonB07** [comment]
Ah, apparently this isn't solved. My tests just failed...

**Comment by DillonB07** [comment]
**I'm incredibly stupid.**
<img width="1315" alt="Screen Shot 2022-01-10 at 5 11 54 pm" src="https://user-images.githubusercontent.com/83948303/148808624-00fbc7cc-73f6-476d-9b61-94d680b5f4bc.png">

**Comment by DillonB07** [comment]
Now it is sorted.
```diff
- return json.dumps(request.json(), indent=4, sort_keys=True)
+ return json.JSONDecoder().decode(json.dumps(request.json(), sort_keys=True))
```
_And I can go back to the `requests` module!_

**Comment by JBYT27** [comment]
Oh, okay! Nice! Let me know if you find any other issues!

**Comment by DillonB07** [comment]
Closed via 9675ec6

**Comment by DillonB07** [comment]
@all-contributors please add @kokonut27 for code, question

*Yes, there's technically none of your code in this repo, but I used your package to help a little*
